pkgs/tools/misc/byobu/default.nix

```diff
@@ -11,26 +16,44 @@ stdenv.mkDerivation rec {
 
   doCheck = true;
 
-  buildInputs = [ python3 perl ];
-  propagatedBuildInputs = [ textual-window-manager ];
-
-  meta = {
+  buildInputs = [ perl pythonEnv makeWrapper vim bc gettext];
+  propagatedBuildInputs = [ textual-window-manager screen pythonEnv ];
+
+  postUnpack = ''
+    substituteInPlace $sourceRoot/usr/bin/byobu-select-profile.in \
+      --replace "/bin/true" "true" \
+      --replace "gettext" "${gettext}/bin/gettext"
+    substituteInPlace $sourceRoot/usr/bin/byobu-export.in \
+      --replace "gettext" "${gettext}/bin/gettext"
+  '';
+
+  postInstall = ''
+    wrapProgram $out/bin/byobu-status-detail --prefix PATH : ${vim}/bin/
+    wrapProgram $out/bin/byobu-ulevel --prefix PATH : ${bc}/bin/
+    wrapProgram $out/bin/byobu-config --set BYOBU_PYTHON ${pythonEnv}/bin/python
+    #for i in $out/bin/*; do
+    #  wrapProgram "$i" --set BYOBU_PYTHON "${pythonEnv}/bin/python" \
+    #    --prefix PATH : ${binPath}
+    #done
```

Comment:
Byobu can be launched with a specific backend through `byobu-screen` and `byobu-tmux`.
To do so, `byobu` checks whether `$0` ends with `byobu-screen` or `byobu-tmux`, but for some reason `$0` is incorrect even though wrapProgram executes programs with the right `$0`.

If it can help, `byobu-screen` and `byobu-tmux` are both symlinks to `byobu-config`.
When everything is wrapped, `byobu-screen` execs `.byobu-screen-wrapped` which points to `byobu-config` which is itself a wrapper for `.byobu-config-wrapped`.
When reading `$0`, `.byobu-config-wrapped` somehow has the wrong filename which makes it start the default backend (tmux).
I'm wondering whether multiple wrappers + symlinks preserve the right filename.

Reply:
you don't have to use `wrapProgram` which by default makes a `.<name>-wrapped`, you can also just do `makeWrapper` and explicitly set what the target filename will be

Reply:
Yes, though wrapProgram already sets the expected filename.
I belive the problem is that one `makeWrapper` is calling another.

Reply:
> you don't have to use wrapProgram which by default makes a .<name>-wrapped, you can also just do makeWrapper and explicitly set what the target filename will be

You were right, I fixed this issue in 3e4b34d